src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -27,12 +28,21 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             IOptions<DurableTaskOptions> options,
             IConnectionStringResolver connectionStringResolver,
             INameResolver nameResolver,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IPlatformInformationService platformInfo = null)
         {
             this.options = options.Value;
             this.nameResolver = nameResolver;
             this.loggerFactory = loggerFactory;
             this.azureStorageOptions = new AzureStorageOptions();
+            this.inConsumption = platformInfo != null ? platformInfo.InConsumption() : false;
+
+            if (this.inConsumption)
+            {
+                this.azureStorageOptions.ControlQueueBufferThreshold = 32;
```

Comment:
32 may be a bit aggressive for all consumption customers, at least for C#. It may be different when we incorporate out-of-proc as well...

I would love to see some fan-out, fan-in perf numbers with both our current default and these new defaults on consumption plan (maybe with 1000 and then 10000?) just to make sure that the new default experience isn't a huge perf drop?

Reply:
Will do, I was planning to provide those before merging.
Also, for clarity, this PR will not consider OOProc performance tuning, but that's coming shortly